src/lib.rs

```diff
@@ -282,11 +282,14 @@ fn create_http_server(
   let context: LemmyContext = federation_config.deref().clone();
   let rate_limit_cell = federation_config.rate_limit_cell().clone();
   let self_origin = settings.get_protocol_and_hostname();
+  let cors_origin_setting = settings.cors_origin;
   // Create Http server with websocket support
   let server = HttpServer::new(move || {
-    let cors_origin = env::var("LEMMY_CORS_ORIGIN");
+    let cors_origin = env::var("LEMMY_CORS_ORIGIN")
+      .ok()
+      .or(cors_origin_setting.clone());
     let cors_config = match (cors_origin, cfg!(debug_assertions)) {
-      (Ok(origin), false) => Cors::default()
+      (Some(origin), false) => Cors::default()
```

Comment:
Actually it would be much nicer if you move all calls to `env::var` into crates/utils so its all in one place. Then this high level code should only have a single value to read, without caring if its coming from env or config.

Reply:
![Screenshot_2023-10-25-09-52-51-748_com termux](https://github.com/LemmyNet/lemmy/assets/930215/bb15e469-66ba-470a-b6d9-04cd36e89777)

There are a couple of these, and the calls are different in each case, some requiring the settings, some not.

Reply:
Okay Im making my own PR with some changes there.